pycyphal/dsdl/__init__.py

```diff
@@ -17,29 +17,145 @@
 The main entity of this module is the function :func:`compile`.
 """
 
+from typing import TypeVar, Type, Sequence, cast, Any, Iterable, Optional, Dict
 from ._compiler import compile as compile  # pylint: disable=redefined-builtin
 from ._compiler import compile_all as compile_all
 from ._compiler import GeneratedPackageInfo as GeneratedPackageInfo
 
-from ._composite_object import serialize as serialize
-from ._composite_object import deserialize as deserialize
+from ._import_hook import install_import_hook as install_import_hook
+import pydsdl
 
-from ._composite_object import get_fixed_port_id as get_fixed_port_id
-from ._composite_object import get_model as get_model
-from ._composite_object import get_class as get_class
-from ._composite_object import get_extent_bytes as get_extent_bytes
+T = TypeVar("T")
```

Comment:
This should be either prefixed with an underscore or moved into a separate module to prevent scope leakage.